devtale/utils.py

```diff
@@ -28,6 +30,12 @@
     "folder-description": FOLDER_SHORT_DESCRIPTION_TEMPLATE,
 }
 
+# budget manager code
+def check_budget(budget_manager, session_id):
+    current_cost = budget_manager.get_current_cost(user=session_id)
+    session_budget = budget_manager.get_total_budget(session_id)
+    if budget_manager.get_current_cost(user=session_id) > budget_manager.get_total_budget(session_id):
+      raise Exception(f"Exceeded the maximum budget for this session - current cost: {current_cost}, session budget: {session_budget}")
```

Comment:
Question: Is the exception intended to run the documentation until the budget limit is reached? In this scenario, would the user need to re-run the documentation with a larger budget? I mean, is the `current_cost` the final cost or the cost of the execution so far?